### PR TITLE
add fault logic

### PR DESCRIPTION
**Pull request by sdimitro**

### CODE REVIEW

File: handlers/handlers.go [hunk 1]
```diff
@@ -13,37 +13,87 @@ type SensorData struct {
 	Name          string    `json:"name"`
 	Location      string    `json:"location"`
 	Serial_Number int       `json:"serial_number"`
-	Temperate     float64   `json:"temperature"`
+	Temperature   float64   `json:"temperature"`
```
Comment (skrzepto) [inline handlers/handlers.go:16]: Good catch :)

File: handlers/handlers.go [hunk 2]
```diff
@@ -66,15 +116,49 @@ func (i *Impl) GET_SensorData_ID(rw http.ResponseWriter, req *http.Request, ps h
...
-func (i *Impl) SensorData(rw http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	if req.Method == "GET" {
-		js, err := json.Marshal(i.sd_table)
-		if err != nil {
-			http.Error(rw, err.Error(), http.StatusInternalServerError)
-			panic(err)
-		}
-		rw.Header().Set("Content-Type", "application/json")
-		rw.Write(js)
+func (i *Impl) GET_SensorData(rw http.ResponseWriter, req *http.Request, _ httprouter.Params) {
```
Comment (skrzepto) [inline handlers/handlers.go:119]: Thanks for cleaning this function up

File: handlers/handlers.go [hunk 3]
```diff
@@ -66,15 +116,49 @@ func (i *Impl) GET_SensorData_ID(rw http.ResponseWriter, req *http.Request, ps h
...
+
+	i.mu.Lock()
+	flts, exist := i.faults[sensor_id]
+	if !exist {
+		rw.WriteHeader(404)
+		rw.Write([]byte("Can't find fault data for the id specified"))
+		i.mu.Unlock()
+		return
+	}
+	js, err := json.Marshal(flts)
```
Comment (skrzepto) [inline handlers/handlers.go:155]: Should we nil the location instead? `i.faults[sensor_id] = nil` ? since this is delete the fault from the table

**Review by skrzepto — CHANGES_REQUESTED** [review]
One small change in the delete_fault function and the rest looks good

File: handlers/handlers.go (hunk 3 above)
Comment (skrzepto) [reply handlers/handlers.go:155]: Nvm i think its `delete(m, k)` where m is the map k is the key

Comment (skrzepto) [reply handlers/handlers.go:155]: yup its delete

```
diff --git a/handlers/handlers.go b/handlers/handlers.go
index cda1ca6..4e70653 100644
--- a/handlers/handlers.go
+++ b/handlers/handlers.go
@@ -153,7 +153,8 @@ func (i *Impl) DELETE_Faults_ID(rw http.ResponseWriter, req *http.Request, ps ht
                return
        }
        js, err := json.Marshal(flts)
-       i.mu.Unlock()
+       delete(i.faults, sensor_id)
+       i.mu.Unlock()

        if err != nil {
                http.Error(rw, err.Error(), http.StatusInternalServerError)
```

Comment (sdimitro) [reply handlers/handlers.go:155]: Sorry I was absent minded. I just gave gave back the faults for the specific sensor. I will change it now. Thanks for catching that

**Comment by sdimitro** [comment]
Updated give it a look and feel free to merge

**Review by skrzepto — APPROVED** [review]

**Pull request merged**